### PR TITLE
[MGDOBR-279] add DELETING status

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-279](https://issues.redhat.com/browse/MGDOBR-279) 

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java [hunk 1]
```diff
@@ -89,7 +89,7 @@ void syncUpdatesFromManager() {
                                                         failure -> failedToSendUpdateToManager(y, failure));
                                     }
                                     if (y.getStatus().equals(BridgeStatus.DELETION_REQUESTED)) { // Bridges to delete
-                                        y.setStatus(BridgeStatus.DELETED);
+                                        y.setStatus(BridgeStatus.DELETING);
```
Comment (sutaakar) [inline shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java:92]: Should it be applied also to `fetchAndProcessProcessorsToDeployOrDelete` below?

Comment (r00ta) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java:92]: yes, thanks

**Review by danielezonca — APPROVED** [review]

**Comment by r00ta** [comment]
> @r00ta DELETING status is not updated for Processor deletion

Hi @Srihari1192 , wdym?

**Comment by Srihari1192** [comment]
> > @r00ta DELETING status is not updated for Processor deletion
> 
> Hi @Srihari1192 , wdym?

@r00ta I mean in [MGDOBR-279](https://issues.redhat.com/browse/MGDOBR-279)  mentioned add Deleting status for Bridge and Processor. Current handled for Bridge Deletion but not for Processor deleting status

**Review by Srihari1192 — APPROVED** [review]

**Comment by r00ta** [comment]
> > > @r00ta DELETING status is not updated for Processor deletion
> > 
> > 
> > Hi @Srihari1192 , wdym?
> 
> @r00ta I mean in [MGDOBR-279](https://issues.redhat.com/browse/MGDOBR-279) mentioned add Deleting status for Bridge and Processor. Currently handled for Bridge Deletion but not for Processor deleting status

As per chat, also the processor is transitioning from AVAILABLE -> DELETING -> DELETED 
Screenshot from 2022-02-02 19-27-02

**Pull request merged**